### PR TITLE
Typescriptify backend

### PR DESCRIPTION
**Pull request by Jikstra**
Lets not merge this before releasing 1.1.0 (the ecc fix release)

### CODE REVIEW

File: src/main/deltachat/chat.ts [hunk 1]
```diff
@@ -16,24 +19,24 @@ module.exports = class DCChat extends SplitOut {
...
   }
 
-  leaveGroup(chatId) {
+  leaveGroup(chatId: number) {
     log.debug(`action - leaving chat ${chatId}`)
     this._dc.removeContactFromChat(chatId, C.DC_CONTACT_ID_SELF)
   }
 
-  setName(chatId, name) {
+  setName(chatId: number, name: any) {
```
Comment (Simon-Laux) [inline src/main/deltachat/chat.ts]: we know that the name is a string don't we?
If it also can be null, which it  probably can **not** in this case, you could do `string|null` in order to be supper correct and to comply with the compiler if we would ever activate "no Implicit null"

Comment (Jikstra) [reply src/main/deltachat/chat.ts]: We should first specify this in deltachat-node and then bubble it up.

Comment (Simon-Laux) [reply src/main/deltachat/chat.ts]: feel free to do so

**Comment by Simon-Laux** [comment]
I'm taking over this pr (continue it)

**Comment by Simon-Laux** [comment]
So the backend is basicaly converted (only the backend in meaning the deltachat folder).
The only thing left to do is review it and resolve some `any`s in login.ts:

> login.ts has still some `any`s that can be resolved such as credentials

**Comment by Simon-Laux** [comment]
Also I have the feeling that I smashed some smaller bugs in the process or at least potential one... :wink:

**Pull request merged**